.github/workflows/build-compiler.yml

```diff
@@ -0,0 +1,119 @@
+name: 'Build compiler'
+
+on:
+  push:
+    branches:
+      - '**'
+    paths:
+      - '.github/workflows/**'
+      - 'examples/**'
+      - 'library/**'
+      - 'source/**'
+      - 'test/**'
+  pull_request:
+    branches:
+      - '**'
+    paths:
+      - '.github/workflows/**'
+      - 'examples/**'
+      - 'library/**'
+      - 'source/**'
+      - 'test/**'
+
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  build-compiler:
+    name: 'Building'
+
+    strategy:
+      matrix:
+        system:
+          - windows-2019
+          - windows-2022
+          - ubuntu-20.04
+          - ubuntu-22.04
+          - macos-11
+          - macos-12
+        node-version:
+          - 16.x
+
+    runs-on: ${{ matrix.system }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - name: Run Npm
+        run: |
+          npm clean-install
+
+      - name: Build Hexa
+        run: |
+          node --trace-uncaught bootstrap.js --define debug=false hexa.json
+          node --trace-uncaught hexa-node.js --define debug=true --define times=false hexa.json
+          node --trace-uncaught hexa-node.js --define debug=false --define times=false hexa.json
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: hexa-node.js-${{ matrix.os }}-${{ github.sha }}
```

Comment:
@acidicMercury8 must be xxx-yyy.js (. extension at the end), not xxx.js.yyy, same for other files

Reply:
@acidicMercury8 does it affect the downloaded file name? Bruh nevermind

Reply:
@PeyTy `name` in upload step is name of the archive, which contains packed artifacts. This is not the name of the artifact itself. It is recommended to use this to differentiate names